### PR TITLE
fix(Rendering): some fixes and performance improvements

### PR DESCRIPTION
**Pull request by martinken**
Clean up some of the rendering interaction code to better
manage animating and to ignore render calls when in the
middle of interacting. Also remove some calls that were
rendering the still image twice.

Also some code cleanup to improve performance some. Still a ways
to go in cases with hundreds of actors.

### CODE REVIEW

**Comment by martinken** [comment]
@jourdain There were enough changes I wanted to get this merged before continuing

File: Sources/Rendering/Core/InteractorStyle/index.js [hunk 1]
```diff
@@ -194,8 +194,6 @@ function vtkInteractorStyle(publicAPI, model) {
   publicAPI.startState = (state) => {
     model.state = state;
     if (model.animationState === States.IS_ANIM_OFF) {
-      // const rwi = model.interactor;
-      // rwi.getRenderWindow().setDesiredUpdateRate(rwi.getDesiredUpdateRate());
       publicAPI.invokeStartInteractionEvent({ type: 'StartInteractionEvent' });
```
Comment (jourdain) [inline Sources/Rendering/Core/InteractorStyle/index.js]: @martinken we could also have that StartInteractionEvent/EndInterationEvent be static object defined at the top of the module.

File: Sources/Rendering/Core/Prop/index.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ function vtkProp(publicAPI, model) {
       Math.max(val.getMTime(), acc)
     , 0));
 
+  publicAPI.getProps = () => publicAPI;
```
Comment (jourdain) [inline Sources/Rendering/Core/Prop/index.js]: There must be a typo here?

Comment (martinken) [reply Sources/Rendering/Core/Prop/index.js]: By default a prop is a prop right? So getProps should by default return itself. Or are you saying it should be [publicAPI] ?

Comment (jourdain) [reply Sources/Rendering/Core/Prop/index.js]: You are right, it is correct. I was expecting another method call and thought you forgot the method.

**Pull request merged**